File: features/steps/csp_steps.js
module.exports = function(){
    this.Given(/^I have a CSP problem$/, function(callback) {
	// express the regexp above with the code you wish you had
	callback.pending();
    });

    this.Given(/^I have a variable 'X' with domain \[(\d+), (\d+), (\d+)\]$/, function(arg1, arg2, arg3, callback) {
	// express the regexp above with the code you wish you had
	callback.pending();
    });

    this.When(/^I solve the problem$/, function(callback) {
	// express the regexp above with the code you wish you had
	callback.pending();
    });

    this.Then(/^I expect solutions$/, function(table, callback) {
	// express the regexp above with the code you wish you had
	callback.pending();
    });

    this.Given(/^I have a variable 'Y' with domain \[(\d+), (\d+)\]$/, function(arg1, arg2, callback) {
	// express the regexp above with the code you wish you had
	callback.pending();
    });

    this.Given(/^I have a constraint: X = Y$/, function(callback) {
	// express the regexp above with the code you wish you had
	callback.pending();
    });

    this.Given(/^I have a constraint: X != Y$/, function(callback) {
	// express the regexp above with the code you wish you had
	callback.pending();
    });

    this.Given(/^I have a constraint: X = Y \+ (\d+)$/, function(arg1, callback) {
	// express the regexp above with the code you wish you had
	callback.pending();
    });
};

File: features/steps/csp_steps.rb
GEM_ROOT = File.expand_path("../../..", __FILE__)
$:.unshift File.join(GEM_ROOT, "lib")

require 'csp'

Given(/^I have a CSP problem$/) do
  @problem = CSP.problem
end

Given(/^I have a variable '(\w+)' with domain \[(.*)\]$/) do |variable, domain|
  pending # express the regexp above with the code you wish you had
end

Given(/^I have a constraint: (.*)$/) do |constraint|
  pending # express the regexp above with the code you wish you had
end

When(/^I solve the problem$/) do
  pending # express the regexp above with the code you wish you had
end

Then(/^I expect solutions$/) do |table|
  # table is a Cucumber::Ast::Table
  pending # express the regexp above with the code you wish you had
end
